libyul/YulName.h

```diff
@@ -18,9 +18,8 @@
 
 #pragma once
 
-#include <libyul/YulString.h>
-
 namespace solidity::yul
 {
+class YulString;
```

Comment:
The advantage is that I am now including `YulName.h` in `ASTForward.h`. This is required if I want to define `FunctionName` in there, as the identifier for a user defined function is a `YulName` (or currently: `YulString`). So by making `YulName.h` forwarding, I don't implicitly include `YulString.h` in `ASTForward.h`.

Reply:
Wouldn't it be more typical for just forward-declare `YulName` there? Or, I guess, redefine since it's just an alias.

In any case, it's fine, I was just curious why it was done this way.

Reply:
Right, redefining is an option :)